RadixWallet/Clients/CloudBackupClient/CloudBackupClient+Live.swift

```diff
@@ -142,20 +143,55 @@ extension CloudBackupClient {
 			try? userDefaults.setLastCloudBackup(result, of: profile)
 		}
 
+		@Sendable
+		func performAutomaticBackup(_ profile: Profile, timeToCheckIfClaimed: Bool) async {
+			let needsBackUp = profile.appPreferences.security.isCloudProfileSyncEnabled && profile.header.isNonEmpty
+			let lastBackup = userDefaults.getLastCloudBackups[profile.id]
+			let lastBackupSucceeded = lastBackup?.result == .success && lastBackup?.saveHash == profile.saveHash
+
+			let shouldBackUp = needsBackUp && !lastBackupSucceeded
+			let shouldCheckClaim = shouldBackUp || timeToCheckIfClaimed
+
+			guard shouldBackUp || shouldCheckClaim else { return }
+
+			let existingRecord = try? await fetchProfileRecord(.init(recordName: profile.id.uuidString))
+
+			let backedUpID = { try? existingRecord.map(getProfileHeader)?.lastUsedOnDevice.id }
+
+			let didTransferBackProfile: Bool
+			if shouldCheckClaim, let id = backedUpID(), await !profileStore.isThisDevice(deviceID: id) {
+				let action = await overlayWindowClient.scheduleFullScreen(.init(root: .claimWallet(.init())))
+				didTransferBackProfile = action == .claimWallet(.didTransferBack)
+			} else {
+				didTransferBackProfile = false
+			}
+
+			guard shouldBackUp || didTransferBackProfile else { return }
+
+			await backupProfileAndSaveResult(existingRecord: existingRecord)
+		}
+
+		let retryBackupInterval: DispatchTimeInterval = .seconds(60)
+		let checkClaimedProfileInterval: TimeInterval = 15 * 60
+
 		return .init(
 			startAutomaticBackups: {
-				let timer = AsyncTimerSequence(every: .seconds(60))
+				let ticks = AsyncTimerSequence(every: retryBackupInterval)
 				let profiles = await profileStore.values()
+				var lastClaimCheck: Date = .distantPast
 
-				for try await (profile, _) in combineLatest(profiles, timer) {
+				for try await (profile, tick) in combineLatest(profiles, ticks) {
 					guard !Task.isCancelled else { return }
-					guard profile.appPreferences.security.isCloudProfileSyncEnabled else { continue }
-					guard profile.header.isNonEmpty else { continue }
-
-					let last = userDefaults.getLastCloudBackups[profile.id]
-					if let last, last.result == .success, last.profileHash == profile.hashValue { continue }
-
-					await backupProfileAndSaveResult(profile)
+					try secureStorageClient.updateIsCloudProfileSyncEnabled(profile.id, .disable)
```

Comment:
Done on every tick?

Reply:
It's way overkill, I really only want to do it when the profileID changes. I added it here at the last minute, but I will find a better place. Actually might be best to put it right outside the loop, because this loop is restarted when you delete the wallet.